client_test/rpc_test.go

```diff
@@ -235,6 +242,22 @@ func checkQueryInfo[T channelInfo](t *testing.T, expected T, fetched T) {
 	}
 }
 
+func checkQueryInfoCollection[T channelInfo](t *testing.T, expected T, expectedLength int, fetched []T) {
+	if len(fetched) != expectedLength {
+		t.Fatalf("expected %d channel infos, got %d", expectedLength, len(fetched))
+	}
+	found := false
+	for _, fetched := range fetched {
+		if cmp.Equal(expected, fetched, cmp.AllowUnexported(big.Int{})) {
+			found = true
+			break
+		}
+	}
+	if !found {
+		panic(fmt.Errorf("did not find info %v in channel infos: %v", expected, fetched))
```

Comment:
is `t.Fatalf` better than `panic`?

Reply:
> is t.Fatalf better than panic?

I think `t.Fatalf` is our preferred option, I think I switched to to `panic` so I could easily identify the error I'll switch it back

. One problem with `t.Fatalf` is that it doesn't give a full stack trace, so if it fails, it's often hard to figure out which call to your test function was the one causing the problem.